Add Nuxt 3 migration project proposal (#2311)
WordPress/openverse: documentation/projects/proposals/nuxt_3_migration/20230604-project_proposal_nuxt_3_migration.md

# 2023-06-02 Project Proposal

**Author**: @olgabulat

## Reviewers

<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->

- [ ] @zackkrida - for previous experience of migrating the project to Nuxt
- [ ] @sarayourfriend - for extensive experience on the frontend

## Project summary

<!-- A brief one or two sentence summary of the project's features -->

Openverse should migrate from Nuxt 2, which is about to reach end-of-life, to
Nuxt 3.

## Goals

<!-- Which yearly goal does this project advance? -->

Developer experience

## Requirements

<!-- Detailed descriptions of the features required for the project. Include user stories if you feel they'd be helpful, but focus on describing a specification for how the feature would work with an eye towards edge cases. -->

- Openverse should run on Nuxt 3.
- All the current functionality should work as expected.
- All the dependencies should be either updated to latest versions that support
  Nuxt 3 or replaced with the ones that support Nuxt 3.

> - (Optional) Nuxt app should use node version 20, and pnpm version 8.

Reviewer: The `pnpm` version upgrade does seem optional to me, but moving off of Node 16 should either be a mandatory requirement of this, either to actually do it or identify the specific things that need to be done beyond Nuxt 3 upgrades (even in passing, to set up a subsequent implementation plan). Node 16 reaching EOL while we're still on it introduces a significant risk of security issues that we should prioritise avoiding.

Author: I'll update it to say:
- Nuxt app should migrate from node version 16 to the active LTS version 18 (or the current version 20, which will become active LTS on 2023-10-24.
- (Optional) Nuxt app should migrate to `pnpm` version 8

> - (Optional) Nuxt app should use node version 20, and pnpm version 8.

Author: ```suggestion

- Nuxt app should migrate from node version 16 to the active LTS version 18 (or the current version 20, which will become active LTS on 2023-10-24).
- (Optional) Nuxt app should migrate to pnpm version 8

```

# 2023-06-02 Project Proposal

**Author**: @olgabulat

## Reviewers

<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->

> - [ ] @zackkrida - for previous experience of migrating the project to Nuxt
> - [ ] @sarayourfriend - for extensive experience on the frontend

Author: ```suggestion
- [x] @zackkrida - for previous experience of migrating the project to Nuxt
- [x] @sarayourfriend - for extensive experience on the frontend
```

## Project summary

<!-- A brief one or two sentence summary of the project's features -->

Openverse should migrate from Nuxt 2, which is about to reach end-of-life, to
Nuxt 3.

## Goals

<!-- Which yearly goal does this project advance? -->

Developer experience

## Requirements

<!-- Detailed descriptions of the features required for the project. Include user stories if you feel they'd be helpful, but focus on describing a specification for how the feature would work with an eye towards edge cases. -->

- Openverse should run on Nuxt 3.
- All the current functionality should work as expected.
- All the dependencies should be either updated to latest versions that support
  Nuxt 3 or replaced with the ones that support Nuxt 3.

> 
> - Nuxt app should migrate from node version 16 to the active LTS version 18 (or the current version 20, which will become active LTS on 2023-10-24).

Reviewer 2: ```suggestion
- Nuxt app should migrate from node version 16 to the active LTS version 18 (or the current version 20, which will become active LTS on 2023-10-24).
```
